metadocs: add poc to single-sourcing/partials (#1230)
redpanda-data/documentation: modules/get-started/pages/release-notes/operator.adoc

== Redpanda Operator v25.2.x

link:https://github.com/redpanda-data/redpanda-operator/blob/release/v25.2.x/operator/CHANGELOG.md[Changelog^]

=== Cluster scope by default

Starting in v25.2, the Redpanda Operator defaults to cluster scope instead of namespace scope. This change provides several benefits:

* **Simplified management**: A single operator instance can manage multiple Redpanda clusters across different namespaces
* **Reduced resource overhead**: No need to deploy separate operator instances for each namespace
* **Centralized upgrades**: Upgrade the operator once to benefit all managed Redpanda clusters
* **Cross-namespace management**: Deploy the operator in a dedicated namespace (such as `redpanda-system`) while managing clusters in application namespaces

==== Migration considerations

If you're upgrading from a previous version that used namespace-scoped operators:

* **No manual steps required**: The Redpanda Operator automatically reconciles existing Redpanda clusters across namespaces
* The upgrade maintains compatibility with existing deployments

> * **New deployments default to cluster scope**: Regardless of which namespace you deploy the Redpanda Operator to (including `default`)

Reviewer: Maybe one point about deleting all operator deployments `helm uninstall` expect the one that after upgrade will be cluster scoped?

To maintain namespace scope, use the `--set 'additionalCmdFlags=["--namespace=<namespace>"]'` flag when installing or upgrading the Redpanda Operator. The `--namespace` flag in the helm command only specifies which namespace to deploy the Redpanda Operator into and does not affect its operational scope.

WARNING: Do not run multiple Redpanda Operators in different scopes (cluster and namespace scope) in the same cluster as this can cause resource conflicts.

==== RBAC requirements

Important RBAC considerations for v25.2+:

* **ClusterRole permissions always required**: Regardless of whether you use cluster or namespace scope, the Redpanda Operator always needs ClusterRole permissions

> * **Automatic configuration**: These permissions are automatically configured when you install the Redpanda Operator with `rbac.createRPKBundleCRs=true` (default)

Reviewer: Please remove this point. `createRPKBundleCRs` is being removed from operator helm chart values.

* xref:upgrade:k-compatibility.adoc[]
* xref:upgrade:k-rolling-upgrade.adoc[]

> NOTE: Redpanda Console v3 is **not yet available when deploying with the Redpanda Operator**. The Redpanda Operator continues to deploy Redpanda Console v2. To try Redpanda Console v3 in Kubernetes, you can xref:deploy:deployment-option/self-hosted/kubernetes/k-production-deployment.adoc[deploy Redpanda using the Redpanda Helm chart] instead of the Redpanda Operator.

Reviewer 2: @RafalKorepta @chrisseto @david-yu  @JakeSCahill .

Is there really no way to set the image tag of Console independent of Redpanda when using the Operator.  I know we dont yet default to Console v3 (already a concern), but Im even more concerned if this is hard-coded.  People have to be able to update to Console V3 when desired

cc @twmb

Reviewer: The image tag is not enough as Console configuration has breaking changes. There is need for migration that helm chart does not provide. 

BTW There is ability to change the image tag.

> NOTE: Redpanda Console v3 is **not yet available when deploying with the Redpanda Operator**. The Redpanda Operator continues to deploy Redpanda Console v2. To try Redpanda Console v3 in Kubernetes, you can xref:deploy:deployment-option/self-hosted/kubernetes/k-production-deployment.adoc[deploy Redpanda using the Redpanda Helm chart] instead of the Redpanda Operator.

Reviewer 3: ```suggestion
NOTE: Redpanda Console v3 is **not yet available when deploying with the Redpanda Operator**. The Redpanda Operator continues to deploy Redpanda Console v2. To try Redpanda Console v3, you can deploy the link:https://artifacthub.io/packages/helm/redpanda-data/console[standalone Redpanda Console Helm chart^] or xref:deploy:deployment-option/self-hosted/kubernetes/k-production-deployment.adoc[deploy Redpanda using the Redpanda Helm chart] instead of the Redpanda Operator.
```

> == Redpanda Operator v25.2.x

Reviewer 2: @JakeSCahill is this going out with Redpanda 25.2? If so its probably worth including in the main 25.2 'what's new" page. cc @micheleRP @david-yu

Reviewer: This PR has hard dependency on https://github.com/redpanda-data/redpanda-operator/pull/974 and future PR release.

link:https://github.com/redpanda-data/redpanda-operator/blob/release/v25.2.x/operator/CHANGELOG.md[Changelog^]

=== Cluster scope by default

Starting in v25.2, the Redpanda Operator defaults to cluster scope instead of namespace scope. This change provides several benefits:

* **Simplified management**: A single operator instance can manage multiple Redpanda clusters across different namespaces
* **Reduced resource overhead**: No need to deploy separate operator instances for each namespace
* **Centralized upgrades**: Upgrade the operator once to benefit all managed Redpanda clusters
* **Cross-namespace management**: Deploy the operator in a dedicated namespace (such as `redpanda-system`) while managing clusters in application namespaces

> * **Simplified RBAC for debug bundles**: The Redpanda Operator now provides all required permissions for `rpk` debug bundle collection by default. The `rbac.createRPKBundleCRs` flag is no longer needed

Reviewer 4: I know you are following the convention that this bulleted list does not have end punctuation, but it's odd that this last one does not have a period when it comes right after a sentence with end punctuation. Maybe give them all periods and then they are consistent.

> * **Simplified management**: A single operator instance can manage multiple Redpanda clusters across different namespaces
> * **Reduced resource overhead**: No need to deploy separate operator instances for each namespace
> * **Centralized upgrades**: Upgrade the operator once to benefit all managed Redpanda clusters
> * **Cross-namespace management**: Deploy the operator in a dedicated namespace (such as `redpanda-system`) while managing clusters in application namespaces
> * **Simplified RBAC for debug bundles**: The Redpanda Operator now provides all required permissions for `rpk` debug bundle collection by default. The `rbac.createRPKBundleCRs` flag is no longer needed

Reviewer 3: ```suggestion
* **Simplified management**: A single operator instance can manage multiple Redpanda clusters across different namespaces.
* **Reduced resource overhead**: No need to deploy separate operator instances for each namespace.
* **Centralized upgrades**: Upgrade the operator once to benefit all managed Redpanda clusters.
* **Cross-namespace management**: Deploy the operator in a dedicated namespace (such as `redpanda-system`) while managing clusters in application namespaces.
* **Simplified RBAC for debug bundles**: The Redpanda Operator now provides all required permissions for `rpk` debug bundle collection by default. The `rbac.createRPKBundleCRs` flag is no longer needed.
```

==== Migration considerations

If you're upgrading from a previous version that used namespace-scoped operators:

> * **No manual steps required**: The Redpanda Operator automatically reconciles existing Redpanda clusters across namespaces

Reviewer 4: ```suggestion
* **No manual steps required**: The Redpanda Operator automatically reconciles existing Redpanda clusters across namespaces.
```

> * **New deployments default to cluster scope**: Regardless of which namespace you deploy the Redpanda Operator to (including `default`)

Reviewer 4: ```suggestion
* **New deployments default to cluster scope**: Regardless of which namespace you deploy the Redpanda Operator to (including `default`).
```

* **Delete extra Redpanda Operator deployments**: After upgrading, ensure only one Redpanda Operator remains in the cluster (the one running in cluster scope). Use `helm uninstall` to remove any other Redpanda Operator deployments from previous namespace-scoped installations.

To maintain namespace scope, use the `--set 'additionalCmdFlags=["--namespace=<namespace>"]'` flag when installing or upgrading the Redpanda Operator. The `--namespace` flag in the helm command only specifies which namespace to deploy the Redpanda Operator into and does not affect its operational scope.

WARNING: Do not run multiple Redpanda Operators in different scopes (cluster and namespace scope) in the same cluster as this can cause resource conflicts.

==== RBAC requirements

Important RBAC considerations for v25.2+:

> * **ClusterRole permissions always required**: Regardless of whether you use cluster or namespace scope, the Redpanda Operator always needs ClusterRole permissions

Reviewer 3: ```suggestion
* **ClusterRole permissions always required**: Regardless of whether you use cluster or namespace scope, the Redpanda Operator always needs ClusterRole permissions.
```

> * **Automatic configuration**: These permissions are automatically configured when you install the Redpanda Operator with `rbac.createRPKBundleCRs=true` (default)

Reviewer 3: ```suggestion
* **Automatic configuration**: These permissions are automatically configured when you install the Redpanda Operator.
```

link:https://github.com/redpanda-data/redpanda-operator/blob/release/v25.2.x/operator/CHANGELOG.md[Changelog^]

=== Cluster scope by default

Starting in v25.2, the Redpanda Operator defaults to cluster scope instead of namespace scope. This change provides several benefits:

* **Simplified management**: A single operator instance can manage multiple Redpanda clusters across different namespaces
* **Reduced resource overhead**: No need to deploy separate operator instances for each namespace
* **Centralized upgrades**: Upgrade the operator once to benefit all managed Redpanda clusters
* **Cross-namespace management**: Deploy the operator in a dedicated namespace (such as `redpanda-system`) while managing clusters in application namespaces
* **Simplified RBAC for debug bundles**: The Redpanda Operator now provides all required permissions for `rpk` debug bundle collection by default. The `rbac.createRPKBundleCRs` flag is no longer needed

==== Migration considerations

If you're upgrading from a previous version that used namespace-scoped operators:

* **No manual steps required**: The Redpanda Operator automatically reconciles existing Redpanda clusters across namespaces

> * The upgrade maintains compatibility with existing deployments

Reviewer 3: ```suggestion
```
